internal/mode/static/handler.go

```diff
@@ -506,7 +540,20 @@ func (h *eventHandlerImpl) nginxGatewayServiceUpsert(ctx context.Context, logger
 		logger.Error(err, "Setting GatewayStatusAddress to Pod IP Address")
 	}
 
-	h.cfg.statusUpdater.UpdateAddresses(ctx, gwAddresses)
+	graph := h.cfg.processor.GetLatestGraph()
+	if graph == nil {
+		return
+	}
+
+	transitionTime := metav1.Now()
+	gatewayStatuses := status.PrepareGatewayRequests(
+		graph.Gateway,
+		graph.IgnoredGateways,
+		transitionTime,
+		gwAddresses,
+		h.latestReloadResult,
```

Comment:
@sjberman 
previously,  the latest reload result was saved in statuses in StatusUpdater, so that 
https://github.com/nginxinc/nginx-gateway-fabric/blob/f3e9b9c1c478ace609126ccd9bd7a9bebbe2472b/internal/mode/static/handler.go#L509 would just update only the IP, retaining the reload result
https://github.com/nginxinc/nginx-gateway-fabric/blob/f3e9b9c1c478ace609126ccd9bd7a9bebbe2472b/internal/framework/status/updater.go#L262

So it doesn't look there was any bug.